AWP-1664: Create Breadcrumb Component

ehsan-atif opened the pull request:
[AWP-1664](https://team-1637335847501.atlassian.net/browse/AWP-1664?atlOrigin=eyJpIjoiMDAwMzY3NGU3NzUxNDUxNWFlODNkMTI0ZTk1NDE1YjMiLCJwIjoiaiJ9)
- create breadcrumb component
- set styles for breadcrumb
- add svg icon for ChevronRight
- integrate breadcrumb component in headers

[AWP-1664]: https://team-1637335847501.atlassian.net/browse/AWP-1664?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ehsan-atif committed:
- create breadcrumb component
- set styles for breadcrumb
- add svg icon for ChevronRight
- integerate breadcrumb componet in headers

gm-global commented on src/components/breadcrums/Breadcrumbs.styles.ts:
```diff
@@ -0,0 +1,41 @@
+  line-height: 150%;
+  letter-spacing: 0.019px;
+  text-decoration: none;
+`;
```
```suggestion
export const BreadcrumbLink = styled.a<{ active?: boolean }>`
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  line-height: 150%;
  letter-spacing: 0.019px;
  text-decoration: none;
  
  color: ${(props) => {
    return props?.active
      ? "var(--primary-300, #115873)"
      : "var(--neutral-200, #9CA3AF)";
  }};
  check: ${(props) => {
    return props.active ? 300 : 300;
  }};
`;
```

gm-global replied:
Can we move dynamic properties at end for better readability?

gm-global commented on src/components/breadcrums/Breadcrumbs.tsx:
```diff
@@ -0,0 +1,53 @@
+      link: PropTypes.string.isRequired,
+    }).isRequired,
+  ).isRequired,
+};
```
Why do we need to do this?

ehsan-atif replied:
it is specifying type of data expected by the component

gm-global commented on src/components/breadcrums/Breadcrumbs.tsx:
```diff
@@ -0,0 +1,53 @@
+export interface breadcrumbItem {
+  label: string;
+  link: string;
+  active?: boolean;
```
Can we set the last item as active instead of passing it as a prop?

gm-global commented on src/components/breadcrums/__docs__/BreadcrumbsExample.tsx:
```diff
@@ -0,0 +1,26 @@
+  );
+};
+
+export default BreadcrumbsExample;
```
We can remove this file because we don't need any action before rendering it

gm-global commented on src/components/header/Header.tsx:
```diff
@@ -18,32 +18,27 @@ export const HeadingComponent: FC<HTMLAttributes<HTMLDivElement>> = ({
         {ButtonSlot && <ButtonSlot />}
       </Container>
-      {BreadcrumbsSlot && <BreadcrumbsSlot />}
+      {breadcrumbItems && <Breadcrumbs items={breadcrumbItems} />}
```
Can we use slot here?

gm-global reviewed (commented):
Can you also resolve console warnings for this?

ehsan-atif committed:
- feedback has been addressed
- rename bredcrumb folder
- set last link as active
- rearange styles for better readability
- delete useless example file

ehsan-atif committed:
- addressed comments
- replace a tag with navlink in breadcrumbs
- pass isactiveLink props to headers and breadcrumbs
- update header and breadcrumb test cases
- create chevron right svg
